service/db/src/main/java/gov/va/vro/service/db/SaveToDbServiceImpl.java

```diff
@@ -30,6 +39,51 @@ public Claim insertClaim(Claim claim) {
     return claim;
   }
 
+  @Override
+  public void insertAssessmentResult(UUID claimId, String evidence, String diagnosticCode)
+      throws NoSuchFieldException {
+
+    ClaimEntity claimEntity =
+        claimRepository
+            .findById(claimId)
+            .orElseThrow(
+                () ->
+                    new NoSuchFieldException(
+                        "Could not match claim ID {" + claimId.toString() + "} in DB."));
+    Map<String, Object> evidenceSummary = new HashMap<>();
+    try {
+      evidenceSummary = objMapper.readValue(evidence, Map.class);
+      if (evidenceSummary.isEmpty()) {
+        throw new NoSuchFieldException("Could not map assessment result evidence summary");
+      }
+    } catch (Exception e) {
+      log.error("Could not map assessment result evidence summary");
+      return;
+    }
+    Map<String, Object> defaultEvidence = new HashMap<>();
+    defaultEvidence.put("evidenceNotAvailable", 0);
+    Map summary = (Map) evidenceSummary.getOrDefault("evidenceSummary", defaultEvidence);
+    AssessmentResultEntity assessmentResultEntity = new AssessmentResultEntity();
+    assessmentResultEntity.setEvidenceCountSummary(summary);
+    ContentionEntity contention = findContention(claimEntity, diagnosticCode);
+    contention.addAssessmentResult(assessmentResultEntity);
+    claimRepository.save(claimEntity);
+  }
+
+  private ContentionEntity findContention(ClaimEntity claim, String diagnosticCode)
+      throws NoSuchFieldException {
+    ContentionEntity resp = new ContentionEntity();
+    for (ContentionEntity contention : claim.getContentions()) {
+      if (contention.getDiagnosticCode().equals(diagnosticCode)) {
+        resp = contention;
+      } else {
+        log.error("Could not find match contention with diagnostic code.");
+        throw new NoSuchFieldException("Could not find match contention with diagnostic code.");
```

Comment:
NoSuchField is not the right exception for this situation. Also, if we throw an exception here, what are the consequences for the route?